src/app/shared/services/api/test-utils.ts

```diff
@@ -1,30 +1,39 @@
+function buildParam(
+  params: URLSearchParams,
+  [paramKey, paramValue]: [string, any],
+): URLSearchParams {
+  if (Array.isArray(paramValue)) {
+    paramValue.forEach((arrayItem, index) => {
+      if (typeof arrayItem === 'object' && arrayItem !== null) {
+        Object.entries(arrayItem).forEach(([objectKey, objectValue]) => {
+          params.append(
+            `${paramKey}[${index}][${objectKey}]`,
+            String(objectValue),
+          );
+        });
+      } else {
+        params.append(`${paramKey}[]`, String(arrayItem));
+      }
+    });
+  } else if (typeof paramValue === 'object' && paramValue !== null) {
+    Object.entries(paramValue).forEach(([objectKey, objectValue]) => {
+      params.append(`${paramKey}[${objectKey}]`, String(objectValue));
+    });
+  } else {
+    params.append(paramKey, String(paramValue));
+  }
+
+  return params;
+}
+
 export function buildUrlWithParams(
   baseUrl: string,
   queryParams: Record<string, any>,
 ): string {
-  const params = new URLSearchParams();
-
-  const buildParam = (key: string, value: any) => {
-    if (Array.isArray(value)) {
-      value.forEach((v, index) => {
-        if (typeof v === 'object' && v !== null) {
-          Object.entries(v).forEach(([k, val]) =>
-            params.append(`${key}[${index}][${k}]`, String(val)),
-          );
-        } else {
-          params.append(`${key}[]`, String(v));
-        }
-      });
-    } else if (typeof value === 'object' && value !== null) {
-      Object.entries(value).forEach(([k, val]) =>
-        params.append(`${key}[${k}]`, String(val)),
-      );
-    } else {
-      params.append(key, String(value));
-    }
-  };
-
-  Object.entries(queryParams).forEach(([key, value]) => buildParam(key, value));
+  const params = Object.entries(queryParams).reduce(
+    (accumulatedParams, entry) => buildParam(accumulatedParams, entry),
+    new URLSearchParams(),
```

Comment:
I think you can make this slightly more concise by passing just `buildParam` in place of `(accumulatedParams, entry) => buildParam(accumulatedParams, entry)`